File: app/src/main/java/com/example/hoavo/karaoke/ui/main/VideoFragment.java
package com.example.hoavo.karaoke.ui.main;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.example.hoavo.karaoke.R;
import com.example.hoavo.karaoke.models.Video;

import java.util.ArrayList;

/**
 * Copyright © 2017 AsianTech inc.
 * Created by rimoka on 21/07/2017.
 */
public class VideoFragment extends Fragment implements ItemVideoListener {
    RecyclerView mRecyclerView;

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, Bundle savedInstanceState) {
        View v = inflater.inflate(R.layout.fragment_video1, container,false);
        mRecyclerView = (RecyclerView) v.findViewById(R.id.recyclerViewVideo);
        new DataFromApi(this).execute("dog");
        return v;
    }

    public void setAdapter(ArrayList<Video> videos) {
        VideoAdapter videoAdapter = new VideoAdapter(videos, this, getContext());
        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(getContext());
        mRecyclerView.setLayoutManager(linearLayoutManager);
        mRecyclerView.setAdapter(videoAdapter);
    }

    @Override
    public void onItemClick(int position) {

    }
}

File: app/src/main/java/com/example/hoavo/kotlin/ui/list_video/VideoFragment.kt
package com.example.hoavo.kotlin.ui.list_video

import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.support.v4.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.example.hoavo.karaoke.R
import com.example.hoavo.kotlin.models.Item
import com.example.hoavo.kotlin.ui.drawer.MainActivity
import com.example.hoavo.kotlin.ui.other.SearchVideo
import com.example.hoavo.kotlin.ui.play_video.PlayerVideoActivity
import kotlinx.android.synthetic.main.fragment_video.view.*


/**
 *  Copyright © 2017 AsianTech inc.
 * Created by rimoka on 31/07/2017.
 */

class VideoFragment : Fragment(), OnItemVideoClickListener {

    private var mContext: MainActivity? = null
    companion object {
        private val mVideoFragment: VideoFragment by lazy { VideoFragment() }
        fun newInstance(): VideoFragment = mVideoFragment
    }

    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val v: View? = inflater?.inflate(R.layout.fragment_video, container, false)
        val searchVideo = SearchVideo()
        searchVideo.initSearch(v, this, context)
        v?.btnSearchVideo?.setOnClickListener {
            searchVideo.search(v)
        }
        return v
    }

    override fun onAttach(context: Context?) {
        super.onAttach(context)
        mContext = context as MainActivity
    }

    override fun onItemClick(type: Int, item: Item) {
        when (type) {
            VideoAdapter.TYPE_VIDEO -> {
                val intent = Intent(mContext, PlayerVideoActivity::class.java)
                intent.putExtra(PlayerVideoActivity.TYPE_ITEM, item)
                startActivity(intent)
            }
            VideoAdapter.TYPE_PLAY_LIST -> {

            }
            VideoAdapter.TYPE_CHANNEL -> {

            }
        }
    }
}
